### PR TITLE
Refactoring and cleanup of Query's _execute() function

### PR DESCRIPTION
**Pull request by rasapetter**
Here are some proposed changes whose purpose it is to improve readability on the rather complex _execute() function in query.js. There were some low hanging fruit in the form of redundant if/else statements and unused variables, however, most of the branch deals with breaking up the function into separate parts.

I'm fairly new to Thinky (and RethinkDB for that matter) so please forgive me if I've made any rookie mistakes.

### CODE REVIEW

**Comment by rasapetter** [comment]
Whoops! Errors. Patch coming up...

**Comment by rasapetter** [comment]
Yup, that's better. It took a while to sort things out due to a misunderstanding early in the branch. Seems OK now - all tests pass and the code should be much easier to read, IMO.

**Comment by neumino** [comment]
Gosh, I'm really sorry @rasapetter, I just found out your pull request while merging #222 -_-

I quickly looked at the changes, they look good to me, and it's actually something I thought about doing, so thanks!
I'll take a more careful look tomorrow or this week end and should merge it.

Thanks for the cleaning :)

**Comment by rasapetter** [comment]
Did you run into any issues here? Or is, perhaps, time an issue? :)

**Comment by neumino** [comment]
Sorry, I got too focused on reqlite. Just checked a few more things, everything looks good.

Thanks a lot @rasapetter for the clean up, rI really appreciate it :)

**Comment by primitive-type** [comment]
Hi @neumino and @rasapetter,

I just wanted to make note of something I came across today. In my codebase, previously, I was calling the ```delete``` method on my models in some tests, as well as using the ```count``` method on a result set (without using ```execute```). After updating thinky to the latest version today, both of these cases started throwing thinky validation errors (correctly, I think) until I added the ```execute``` method. 

In other words, not sure if you were already aware, but this PR seems to have improved the strictness of ```run``` vs. ```execute```. At least, that is my best guess as to what happened since these parts of my code haven't changed in weeks.

Thanks!

**Comment by neumino** [comment]
It is actually not related to this PR, but to https://github.com/neumino/thinky/pull/222
And the `npm publish` command somehow didn't release 2.0.4
